packages/google_maps_flutter/google_maps_flutter/example/lib/map_click.dart

```diff
@@ -31,9 +31,9 @@ class _MapClickBody extends StatefulWidget {
 class _MapClickBodyState extends State<_MapClickBody> {
   _MapClickBodyState();
 
-  GoogleMapController mapController;
-  LatLng _lastTap;
-  LatLng _lastLongPress;
+  GoogleMapController? mapController;
```

Comment:
should this be late similar to `animate_camera.dart`?

Reply:
`build` kicks off async maps creation, and also uses `mapController` if present, so it does need to be nullable.

I went back and took a closer look at the `late` cases, and all but one were actually wrong, and had a (latent) bug where there was a race condition where someone could potentially press a control before the map controller was actually created. (I was originally fooled because one of them has a boolean for whether creation has completed and gates showing the controls on that boolean, so `late` is correct, and I missed that others that used the controller unconditionally in callbacks didn't have the same structure.)

I think it's not ideal that there are at least three different patterns for handling the async construction in these example files, but don't want to make changing that part of nullability migration.

Reply:
ah I see. Yeah, I didn't have that context. Thanks